api/server/controllers/userController.js

```diff
@@ -1,298 +1,239 @@
 // Import user model
 User = require("../models/userModel");
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
 
-// Receives details about the new user and creates a new user
-// on the DataBase
-// Returns an error msg
-exports.register = function (req, res) {
-  var user = new User();
-  user.username = req.body.username.toLowerCase();
-  user.email = req.body.email;
-  user.phone = req.body.phone;
-  user.password = req.body.password;
-  user.displayName = req.body.displayName;
-  user.save(function(err){
-    if(err) {
+module.exports = {
+  register: async (req, res) => {
+    const user = await User.findOne({ username: req.body.cred });
```

Comment:
What does `req.body.cred` stand for? Where will its value be read from?

Reply:
cred stands for credential, I changed it because the login route works with either username or email.

Reply:
the attribute on the frontend object/form will have to be updated

Reply:
I see, that sounds clear. I didn't know both username and email are possible